rtgui/toolpanelcoord.cc

```diff
@@ -336,6 +336,24 @@ void ToolPanelCoordinator::panelChanged (rtengine::ProcEvent event, const Glib::
         }
     }
 
+    int tr = TR_NONE;
+    if (params->coarse.rotate == 90) {
+        tr = TR_R90;
+    } else if (params->coarse.rotate == 180) {
+        tr = TR_R180;
+    } else if (params->coarse.rotate == 270) {
+        tr = TR_R270;
+    }
+
+    // Update "on preview" geometry
+    if (event == rtengine::EvPhotoLoaded || event == rtengine::EvProfileChanged || event == rtengine::EvHistoryBrowsed || event == rtengine::EvCTRotate) {
+        // updating the "on preview" geometry
+        int fw, fh;
+        rtengine::ImageSource *ii = (rtengine::ImageSource*)ipc->getInitialImage();
```

Comment:
You're right. I've found an even better solution, replacing :
```
rtengine::ImageSource *ii = (rtengine::ImageSource*)ipc->getInitialImage()->getImageSource();
ii->getFullSize (fw, fh, tr);
```
by
```
ipc->getInitialImage()->getImageSource()->getFullSize (fw, fh, tr);
```
Ok, it doesn't check for NULL pointers, but we're assuming that it's non-NULL.